Cookbook/Technical-Documents/CRP-Bot.md

```diff
@@ -0,0 +1,101 @@
+  - If a release with that exact name already exists in a board, the bot will reuse it instead of creating a new one.
+* Add that newly created version to the "Fix Version" field of the tickets found in the CHANGELOG
+
+You can find an example of [a CRP ticket dedicated to an iOS release for Babylon UK 4.4.0 here](https://babylonpartners.atlassian.net/browse/CRP-4578).
+
+> Note: for documentation about how the CRP is _implemented_ in the code of our Stevenson bot, see [this companion document](CRP-Bot-ImplementationDetails.md)._
+
+## Projects whitelist
+
+To avoid accidentally creating JIRA versions on boards others than the ones we own, or messing with boards that don't follow the same schema/template as the one expected by the bot, the CRP bot has a whitelist of boards on which it will be allowed to create JIRA versions and set Fix Versions fields.
+
+* This provides some security against any `[XXX-NNN]` that might be contained in a commit message to reference a JIRA ticket on some board that we don't own (e.g. a ticket from BackEnd), for which we don't want our bot to act
+* This also allows us to not whitelist some of our boards, for which we ideally would want the CRP bot to work on, but which sadly don't use the official JIRA template that the bot expects – which means that they might not have the same fields in their JIRA tickets, or have different field IDs in the JIRA API for some fields (and sending requests to update a given Field ID would mean accidentally update the wrong field)
+
+This is why some of the boards are still not whitelisted. This is tracked in [IOSP-101](https://babylonpartners.atlassian.net/browse/IOSP-101)
+
+Note that the whitelist only affects the creation of JIRA releases and update of "Fix Version" field on those boards. A link to the tickets will still be included in the description/body of CRP ticket (listing all tickets found in the changelog messages) even for tickets belonging to a non-whitelisted board.
+
+## Running the CRP bot
+
+Currently, the CRP bot can be triggered by a slack command `/crp <platform> <branch>` (e.g. `/crp ios release/babylon/4.4.0`) in the `#ios-launchpad` slack channel.
+
+In the near future, it will likely be automatically triggered by the script doing the release cut every second Friday.
+
+## JIRA API limitations
+
+Given some quota limitations with the JIRA API – which makes JIRA not respond to some requests if we send too many requests in parallel – the CRP bot uses some throttling mechanism to only make API calls to set the "Fix Version" field of each ticket by batches, instead of all at once. This improves the behavior of the JIRA API, but that API still sometimes fail with timeout on some requests – often the JIRA server seem to have processed the request and updated the Fix Version field as requested by the API call, but never returns an HTTP response to the client to let it know it succeeded.
```

Comment:
```suggestion
Given some quota limitations with the JIRA API – which makes JIRA not respond to some requests if we send too many requests in parallel – the CRP bot uses a throttling mechanism to only make API calls to set the "Fix Version" field of each ticket by batches instead of all at once. This improves the behavior of the JIRA API, but that API still sometimes fail with timeout on some requests – often the JIRA server seem to have processed the request and updated the Fix Version field as requested by the API call, but never returns an HTTP response to the client to let it know it succeeded.
```
Unsure if we should remove this last bit: 
```
often the JIRA server seem to have processed the request and updated the Fix Version field as requested by the API call, but never returns an HTTP response to the client to let it know it succeeded.
```
since you've already explained it not long ago (a couple of paragraphs above I think)

Reply:
Above, in `SlowClient`

Reply:
Ah, above as in "above in the PR" not "above in the document". The mention around `SlowClient` is in the technical implementation document – which people will probably not read if they are only interested in usage doc, which is why the explanation is kinda repeated here 😉